src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -370,13 +372,20 @@ private UnfilteredRowIterator queryStorageAndFilter(PrimaryKey key)
                 return null;
 
             lastKey = key;
+            long startTimeNanos = Clock.Global.nanoTime();
 
             try (UnfilteredRowIterator partition = queryController.queryStorage(key, executionController))
             {
                 queryContext.partitionsRead++;
                 queryContext.checkpoint();
 
-                return applyIndexFilter(key, partition, filterTree, queryContext);
+                UnfilteredRowIterator filtered = applyIndexFilter(key, partition, filterTree, queryContext);
+
+                // Note that we record the duration of the read after post-filtering, which actually 
+                // materializes the rows from disk.
+                queryController.tableQueryMetrics.postFilteringReadLatency.update(Clock.Global.nanoTime() - startTimeNanos, TimeUnit.NANOSECONDS);
```

Comment:
Just to clarify, this is recording the per-partition post-filtering latency. Do we want to change the metric name to reflect this. Something like `postFilteringPartitionReadLatency`?

Reply:
I went back and forth on this a little bit. Especially since I'd like to batch row reads into the same partition read sooner rather than later, and even when we read single rows, we're issuing a partition read, it felt unnecessary to qualify the already long name even further. WDYT?